DuckDuckGo/IconManager.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  IconManager.swift
+//  DuckDuckGo
+//
+//  Copyright © 2020 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import UIKit
+
+class IconManager {
+
+    static var shared = IconManager()
+
+    var isIconChangeSupported: Bool {
+        if #available(iOS 10.3, *) {
+            return UIApplication.shared.supportsAlternateIcons
+        } else {
+            return false
+        }
+    }
+
+    enum IconManagerError: Error {
+        case changeNotSupported
+    }
+
+    func changeApplicationIcon(_ icon: Icon) throws {
+        if #available(iOS 10.3, *), isIconChangeSupported {
+            let alternateIconName = icon != Icon.defaultIcon ? icon.rawValue : nil
```

Comment:
Maybe we could ignore situation when user selects icon that is already selected?

Reply:
Good idea ✅